src/receive_imf.rs

```diff
@@ -2391,12 +2376,31 @@ async fn apply_group_changes(
                     &new_members,
                 )
                 .await?;
-                chat_contacts = new_members;
                 send_event_chat_modified = true;
             }
         }
     }
 
+    let new_chat_contacts = HashSet::<ContactId>::from_iter(
+        chat::get_chat_contacts(context, chat_id)
+            .await?
+            .iter()
+            .copied(),
+    );
+
+    // These are for adding info messages about implicit membership changes, so they are only
+    // filled when such messages are needed.
```

Comment:
I don't understand what "so they are only filled when such messages are needed" means? I mean, it's not true, they are even filled if no implicit-membership-change-messages are needed, and in this case, the unnecessary entries are removed again with `added_ids.remove(&added_id)` and `removed_ids.remove(&removed_id);`?

Maybe just remove it

```suggestion
    // These are for adding info messages about implicit membership changes
```